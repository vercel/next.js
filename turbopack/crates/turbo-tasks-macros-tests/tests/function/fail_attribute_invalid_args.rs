#![feature(arbitrary_self_types)]
#![feature(arbitrary_self_types_pointers)]

use turbo_tasks::{ResolvedVc, Vc};

#[turbo_tasks::value(transparent)]
struct IntegersVec(Vec<ResolvedVc<u32>>);

#[turbo_tasks::function(invalid_argument)]
fn return_contains_resolved_vc() -> Vc<IntegersVec> {
    Vc::cell(Vec::new())
}

fn main() {
    // the macro should be error-tolerent and this function should still be created
    // despite the earlier compilation error, so this line should not also error
    let _ = return_contains_resolved_vc();
}
