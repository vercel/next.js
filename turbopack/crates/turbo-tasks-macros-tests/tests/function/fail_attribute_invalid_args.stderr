error: unexpected token, expected one of: "fs", "network", "non_local_return", "operation", or "local_cells"
 --> tests/function/fail_attribute_invalid_args.rs:9:25
  |
9 | #[turbo_tasks::function(invalid_argument)]
  |                         ^^^^^^^^^^^^^^^^
